rasa/server.py

```diff
@@ -744,6 +866,8 @@ async def append_events(request: Request, conversation_id: Text) -> HTTPResponse
             "to the state of a conversation.",
         )
 
+        validate_events_in_request_body(request)
```

Comment:
can't this replace `validate_request_body`?

Reply:
I've tried and most of the tests in `server.py` fail because the `request.body` didn't contain events, unless I do a check ```
if "events" in request.body:
    validate_events_in_request_body(request)
```
Would that be a good idea? 🤔

Reply:
mhm, can you point me to an example of a failing check?

Reply:
Ah I tried locally and I didn't make a note, but I vaguely remember these 🤔 :

- https://github.com/RasaHQ/rasa/blob/fa0fc6119b214c2f9be39486d53bcacf487892f0/tests/test_server.py#L445
- https://github.com/RasaHQ/rasa/blob/fa0fc6119b214c2f9be39486d53bcacf487892f0/tests/test_server.py#L474
- https://github.com/RasaHQ/rasa/blob/fa0fc6119b214c2f9be39486d53bcacf487892f0/tests/test_server.py#L510
- https://github.com/RasaHQ/rasa/blob/fa0fc6119b214c2f9be39486d53bcacf487892f0/tests/test_server.py#L606
- https://github.com/RasaHQ/rasa/blob/fa0fc6119b214c2f9be39486d53bcacf487892f0/tests/test_server.py#L772

Reply:
Yeah, that makes sense. They receive completely different payloads. So they should continue to use the old validation function `validate_request_body`. Only endpoints which receive events should use `validate_events_in_request_body`

Reply:
Can we then drop `validate_request_body` for .e.g. `append_events`, `replace_events` and `validate_events_in_request_body`?

Reply:
By that you mean adding the check if `request.body` is not empty in `validate_events_in_request_body`? Then removing this from the 3 routes (`append_events`, `replace_events`, `tracker_predict`)?

Reply:
> adding the check if request.body is not empty in validate_events_in_request_body?

Isn't that covered by jsonschema validation on `request.json`? So you might not even need to validate the presence of a body separately.

Reply:
Hmm, according to this [post](https://stackoverflow.com/questions/65999411/json-schema-validate-non-empty-body), it seems json-schema doesn't cover validation over an empty instance...unless i'm reading this completely wrong 🙃 Would an empty `request.json` be an empty `{ }` in which case according to this post, validation will fail and I can remove the `validate_request_body` as requested?

Reply:
Either that or just `None`. I don't know - you'd probably need to try out (or write a test) for it.

Reply:
Added some tests => Empty `request.json` evaluates to `None` and the schema validation fails, so all good to remove `validate_request_body`; the only drawback is that the error message won't be as informative for an empty request body as it was with `validate_request_body`.